### ISSUE TITLE
Can I get `ParserRuleContext` from `ASTNode`?

### ISSUE THREAD

**Issue by oovm** [antlr/antlr4-intellij-adaptor#38]
`ANTLRPsiNode` created via `createElement` seems unable to obtain `ParserRuleContext`.

```kotlin
ParserDefinition.createElement(node: ASTNode): PsiElement =  ANTLRPsiNode(node)
```

I want to get child nodes by judging `ParserRuleContext`'s type.

But the child node is also a `ParserRuleContext`, I don’t know how to convert it to a `PsiElement`

**Comment by ris58h** [comment]
You can obtain `PsiElement` from `ASTNode` via `getPsi` method. In Kotlin it's just `psi` property. So `node.psi` will do the trick.

**Comment by bjansen** [comment]
Right, normally you would walk a PSI tree using `psiElement.getChildren()`, then call `child.getNode().getElementType()` to know whick kind of node it is. You can't access an instance of `ParserRuleContext`, but you can do something like this:

```java
var rules  = PSIElementTypeFactory.getRuleIElementTypes(myLanguage);

if (child.getNode().getElementType().equals(rules.get(MyParser.RULE_myRule))) {
    // `child` was created from a parser rule named `myRule`
}
```
